### DISCUSSION TITLE
Get data using XMLHttpRequest request?

### DISCUSSION THREAD

**Discussion by craigmichaelmartin** [TarekRaafat/autoComplete.js#286] [Q&A]
Is there a way to use the older (better supported) XMLHttpRequest function to get results for the autocomplete?

It looks like the current API depends on the data.src function returning an array, or a promise that resolves to an array. In an effort to support as many browsers as possible, I was hoping to using XMLHttpRequest rather than fetch and promises. To support this, the data.src function would need to accept as a parameter a callback function which the data from the API would be passed to - rather than them (or a promise which resolves to them) being returned.

I'm guess this library pre-existed standardized promises, so was wondering if this was/is part of the API, just not shown anymore. Or perhaps the internals of the library now depend on an environment which supports promises, making this a moot effort.

Thanks so much for the great library!!

**Comment by folknor** [comment]
Yes, https://github.com/TarekRaafat/autoComplete.js/issues/255#issuecomment-875462700

**Reply by craigmichaelmartin** [reply]
Thanks for the quick response! This still is the path of using promises. I was wondering if there was support for a path which doesn't involve promises.

**Reply by folknor** [reply]
Ah, I see.

Well then you'll probably want to investigate pointing `data.src` to a non-function structure that supports `forEach` that you wrap in a `Proxy` to detect access.

Even still the autoComplete.js code internally uses `await` earlier in the call chain, so you can't avoid it.

Is that what you wanted? Is this question also asking for a change to autoComplete.js internals?
EDIT: If so this might be useful to the maintainer (though I've not read it, I just found it), unless he makes it a flag somewhere; https://lsm.ai/posts/7-ways-to-detect-javascript-async-function/

Or are you actually looking for coding help?

How does a JS engine that doesn't know Promises behave when its parser encounters the `await` keyword?

**Reply by craigmichaelmartin** [reply]
Gotcha, thanks! Not looking for coding help - just wondering if the API supported it. It sounds like a moot point anyway if promises are used internally. Not looking to change autoComplete internals. Thanks for your feedback!
